File: multi/src/main/java/com/wyyu/multi/cell/IHolderCell.java
package com.wyyu.multi.cell;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Created by wyyu on 2019-09-27.
 **/

public interface IHolderCell {

    void cacheCell(int index, @NonNull Object item);

    void bindParams(int index, @Nullable Object... params);

    void updateCell(int updateType, Object... params);
}

File: multi/src/main/java/com/wyyu/multi/cell/IHolderCell.kt
package com.wyyu.multi.cell

import android.view.View
import androidx.annotation.LayoutRes

/**
 * Created by wyyu on 2021/7/29.
 **/

interface IHolderCell {

    @LayoutRes
    fun getHolderLayout(): Int

    fun onCreateView(itemView: View)

    fun onBindCell(position: Int, itemData: Any?, vararg params: Any?)

    fun onUpdateCell(updateType: Int, vararg params: Any?)
}